### PR TITLE
Ignoring _version_ field for glob patterns on export since we typically do not want that returned unless specifically added as an fl param

### PR DESCRIPTION
**Pull request by justinrsweeney**

### CODE REVIEW

File: solr/core/src/test/org/apache/solr/handler/export/TestExportWriter.java [hunk 1]
```diff
@@ -1333,6 +1333,9 @@ public void testGlobFields() throws Exception {
     assertFalse(
         "Fields without docvalues and useDocValuesAsStored should not be returned",
         jq.contains("\"sortabledv\""));
+    assertFalse(
+        "Fields without docvalues and useDocValuesAsStored should not be returned",
+        jq.contains("\"_version_\""));
```
Comment (magibney) [inline solr/core/src/test/org/apache/solr/handler/export/TestExportWriter.java:1338]: Should this error message be changed?

**Review by magibney — APPROVED** [review]
I can't imagine a situation in which we'd want the `_version_` field, so this seems good to me.

If/when upstreaming, it's possible there'd be concern about parity with glob field handling in `SolrReturnFields` (which I think _does_ include version field?).

I guess the alternative would be to exclude the `_version_` field in whatever code consumes the output of the export command? Is there some challenge there that would make it preferable to exclude `_version_` here?

**Comment by justinrsweeney** [comment]
> I can't imagine a situation in which we'd want the `_version_` field, so this seems good to me.
> 
> If/when upstreaming, it's possible there'd be concern about parity with glob field handling in `SolrReturnFields` (which I think _does_ include version field?).
> 
> I guess the alternative would be to exclude the `_version_` field in whatever code consumes the output of the export command? Is there some challenge there that would make it preferable to exclude `_version_` here?

We don't really have a single place where we parse the result into json to make it easy to remove a field for all callers. On the Go side we generally pass the data around as bytes. We could decode, remove the _version_ and turn back to bytes but these seemed easier on our fork.

For upstreaming I think there are a number of improvements to make to be more consistent on glob pattern/selected field handling so I think we can just reconcile these approaches when this eventually makes it upstream, which will likely take quite a bit more time.

**Comment by magibney** [comment]
That all makes perfect sense. Thanks for the extra clarification!

**Pull request merged**